### ISSUE TITLE
Can't build Crest with latest setuptools

### ISSUE THREAD

**Issue by rkd-msw** [Metaswitch/crest#292]
#### Symptoms

```
$ make env
# Set up a fresh virtual environment
virtualenv --setuptools --python=/usr/bin/python /home/ubuntu/crest/_env
Already using interpreter /usr/bin/python
New python executable in /home/ubuntu/crest/_env/bin/python
Installing setuptools, pip...done.
/home/ubuntu/crest/_env/bin/easy_install -U "setuptools>=3.3"
Searching for setuptools>=3.3
Reading https://pypi.python.org/simple/setuptools/
Best match: setuptools 25.1.0
Downloading https://pypi.python.org/packages/29/c1/33ab781160aede84140523f2fd0685b0bbe46b2d0b7aa9b033a9665cabc4/setuptools-25.1.0.zip#md5=3a2ef546a9f30c41bfc74b6df2e7be5b
Processing setuptools-25.1.0.zip
Writing /tmp/easy_install-8LuzCi/setuptools-25.1.0/setup.cfg
Running setuptools-25.1.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-8LuzCi/setuptools-25.1.0/egg-dist-tmp-vfiLYS
warning: no files found matching '*' under directory 'setuptools/_vendor'
Adding setuptools 25.1.0 to easy-install.pth file
Installing easy_install script to /home/ubuntu/crest/_env/bin
Installing easy_install-2.7 script to /home/ubuntu/crest/_env/bin

Installed /home/ubuntu/crest/_env/lib/python2.7/site-packages/setuptools-25.1.0-py2.7.egg
Processing dependencies for setuptools>=3.3
Finished processing dependencies for setuptools>=3.3
/home/ubuntu/crest/_env/bin/easy_install distribute
Searching for distribute
Reading https://pypi.python.org/simple/distribute/
Downloading https://pypi.python.org/packages/5f/ad/1fde06877a8d7d5c9b60eff7de2d452f639916ae1d48f0b8f97bf97e570a/distribute-0.7.3.zip#md5=c6c59594a7b180af57af8a0cc0cf5b4a
Best match: distribute 0.7.3
Processing distribute-0.7.3.zip
Writing /tmp/easy_install-RBM4Ta/distribute-0.7.3/setup.cfg
Running distribute-0.7.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-RBM4Ta/distribute-0.7.3/egg-dist-tmp-XqHrUH
warning: install_lib: 'build/lib.linux-x86_64-2.7' does not exist -- no Python modules to install

Moving distribute-0.7.3-py2.7.egg to /home/ubuntu/crest/_env/lib/python2.7/site-packages
Adding distribute 0.7.3 to easy-install.pth file

Installed /home/ubuntu/crest/_env/lib/python2.7/site-packages/distribute-0.7.3-py2.7.egg
Processing dependencies for distribute
Finished processing dependencies for distribute
# Generate .egg files for crest, homer, homestead_prov
/home/ubuntu/crest/_env/bin/python setup_crest.py bdist_egg -d .crest-eggs
Traceback (most recent call last):
  File "setup_crest.py", line 57, in <module>
    packages=find_packages('src', include=['metaswitch', 'metaswitch.crest', 'metaswitch.crest.*']),
TypeError: find_packages() got an unexpected keyword argument 'include'
make: *** [/home/ubuntu/crest/_env/.eggs_installed] Error 1
```

#### Impact

Can't build Crest.

#### Release and environment 

Any recent release.

#### Steps to reproduce

Run `make clean` then `make env`.

**Comment by rkd-msw** [comment]
It looks like the problem is with setuptools 2.5, which has removed the `include` argument from find_packages. Oddly, I can see this with `help(find_packages)` but not in the Github source code...

**Comment by rkd-msw** [comment]
No, that's not right - it's that setuptools isn't installing properly, so we're still using the 2.2 version which doesn't have the right arguments. Installing it through `pip` works.

**Comment by rkd-msw** [comment]
This only affects Crest - other Python components install setuptools 2.2 and don't use find_packages. We should be consistent.

**Comment by rkd-msw** [comment]
Sadly, trying to be consistent like that causes the Crest build to fail in this way:

```
# Download the additional egg files homestead_prov depends upon
/home/ubuntu/crest/_env/bin/easy_install -zmaxd .homestead_prov-eggs/ .homestead_prov-eggs/*.egg
Traceback (most recent call last):
  File "/home/ubuntu/crest/_env/bin/easy_install", line 9, in <module>
    load_entry_point('setuptools==25.1.0', 'console_scripts', 'easy_install')()
  File "/home/ubuntu/crest/_env/local/lib/python2.7/site-packages/pkg_resources.py", line 353, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/home/ubuntu/crest/_env/local/lib/python2.7/site-packages/pkg_resources.py", line 2321, in load_entry_point
    return ep.load()
  File "/home/ubuntu/crest/_env/local/lib/python2.7/site-packages/pkg_resources.py", line 2048, in load
    entry = __import__(self.module_name, globals(),globals(), ['__name__'])
  File "build/bdist.linux-x86_64/egg/setuptools/__init__.py", line 11, in <module>
  File "build/bdist.linux-x86_64/egg/setuptools/extern/__init__.py", line 1, in <module>
ImportError: No module named extern
make: *** [/home/ubuntu/crest/_env/.eggs_installed] Error 1
```

I'll stick with the fix in #293.

**Comment by rkd-msw** [comment]
Fixed by #293.
